components/tidb_query_executors/src/runner.rs

```diff
@@ -471,6 +471,9 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
                     &mut ctx,
                 )?
             };
+            if chunk.has_rows_data() {
```

Comment:
Though I prefer to use following approach.  ForwardScanner maintains the statistics (via the Cursor interface), but it's lost before returning to internal_handle_request() in the test.  This patch is ok to me at present. 
<img width="735" alt="image" src="https://user-images.githubusercontent.com/97348524/184097265-cbbb0ade-b56f-4cc6-9f5d-ef512e864722.png">

Reply:
note here the limiter is foreground limiter---that is to limits actual response size back to TiDB, not the IO limiter.